src/Slot-Examples/DerivedSlot.class.st

```diff
@@ -0,0 +1,29 @@
+"
+A derived slot is a virtual slot that stores its value in a hidden base slot.
+"
+Class {
+	#name : #DerivedSlot,
```

Comment:
It's not super clear the difference between a `DerivedSlot` and a `ComposedSlot`/`SlotDecorator`...

Reply:
Yes, there is the question if I should do all the composition stuff outside the image for now... it is exploratory and will change as the examples are implemented.

Reply:
I will keep DerivedSlot as a superclass for now as there is code shared, I hope to find a better name as part of the composition work.